dash_auth/group_protection.py

```diff
@@ -212,3 +296,51 @@ def wrap(*args, **kwargs):
         return wrap
 
     return decorator
+
+
+def filter_kwargs(expected_keys, **kwargs):
+    return {
+        key: value for key, value in kwargs.items() if key in expected_keys
+    }
+
+
+expected_keys = [
+    "unauthenticated_output",
+    "missing_permissions_output",
+    "groups",
+    "groups_key",
+    "groups_str_split",
+    "check_type",
+    "group_lookup",
+    "restricted_users",
+    "restricted_users_lookup",
+    "user_session_key",
+]
+
+
+def protect_layouts(**kwargs) -> str:
+    if "pages_folder" in dash.get_app().config:
+        for pg in dash.page_registry.values():
+            filtered_kwargs = filter_kwargs(expected_keys, **{**kwargs, **pg})
+            if filtered_kwargs.get("unauthenticated_output") is None:
+                filtered_kwargs["unauthenticated_output"] = html.Div(
+                    "you do not have access to this content"
+                )
+            if kwargs.get("public_routes"):
+                if isinstance(kwargs.get("public_routes"), list):
+                    if not (
+                        pg["path"] in kwargs.get("public_routes")
+                        or pg.get("path_template")
+                        in kwargs.get("public_routes")
+                    ):
+                        pg["layout"] = protected(**filtered_kwargs)(
+                            pg["layout"]
+                        )
+                elif not (
+                    kwargs.get("public_routes").test(pg.get("path_template"))
+                    or kwargs.get("public_routes").test(pg["path"])
+                ):
+                    pg["layout"] = protected(**filtered_kwargs)(pg["layout"])
+            else:
+                pg["layout"] = protected(**filtered_kwargs)(pg["layout"])
+        return "your layouts are now protected"
```

Comment:
This can go

Reply:
What about this "can go"? The return statement?

Reply:
Yeah it's a bit strange to have a return for this function